mne/io/boxy/boxy.py

```diff
@@ -50,73 +50,104 @@ class RawBOXY(BaseRaw):
         ###load and read data to get some meta information###
         ###there is alot of information at the beginning of a file###
         ###but this only grabs some of it###
-        filetype = 'parsed'
-        with open(files['001'],'r') as data:
-            for line_num,i_line in enumerate(data,1):
-                if '#DATA ENDS' in i_line:
-                    end_line = line_num - 1
-                    break
-                if 'Detector Channels' in i_line:
-                    detect_num = int(i_line.rsplit(' ')[0])
-                elif 'External MUX Channels' in i_line:
-                    source_num = int(i_line.rsplit(' ')[0])
-                elif 'Auxiliary Channels' in i_line:
-                    aux_num = int(i_line.rsplit(' ')[0])
-                elif 'Waveform (CCF) Frequency (Hz)' in i_line:
-                    ccf_ha = float(i_line.rsplit(' ')[0])
-                elif 'Update Rate (Hz)' in i_line:
-                    srate = float(i_line.rsplit(' ')[0])
-                elif 'Updata Rate (Hz)' in i_line:
-                    srate = float(i_line.rsplit(' ')[0])
-                elif '#DATA BEGINS' in i_line:
-                    start_line = line_num
-                elif 'exmux' in i_line:
-                    filetype = 'non-parsed'
-             
+        detect_num = []
+        source_num = []
+        aux_num = []
+        ccf_ha = []
+        srate = []
+        start_line = []
+        end_line = []
+        filetype = ['parsed' for i_file in files['*.[000-999]*']]
+        for file_num,i_file in enumerate(files['*.[000-999]*'],0):
+            with open(i_file,'r') as data:
+                for line_num,i_line in enumerate(data,1):
+                    if '#DATA ENDS' in i_line:
+                        end_line.append(line_num - 1)
+                        break
+                    if 'Detector Channels' in i_line:
+                        detect_num.append(int(i_line.rsplit(' ')[0]))
+                    elif 'External MUX Channels' in i_line:
+                        source_num.append(int(i_line.rsplit(' ')[0]))
+                    elif 'Auxiliary Channels' in i_line:
+                        aux_num.append(int(i_line.rsplit(' ')[0]))
+                    elif 'Waveform (CCF) Frequency (Hz)' in i_line:
+                        ccf_ha.append(float(i_line.rsplit(' ')[0]))
+                    elif 'Update Rate (Hz)' in i_line:
+                        srate.append(float(i_line.rsplit(' ')[0]))
+                    elif 'Updata Rate (Hz)' in i_line:
+                        srate.append(float(i_line.rsplit(' ')[0]))
+                    elif '#DATA BEGINS' in i_line:
+                        start_line.append(line_num)
+                    elif 'exmux' in i_line:
+                        filetype[file_num] = 'non-parsed'
+                        
         # Extract source-detectors
         ###set up some variables###
-        chan_num = []
+        chan_num_1 = []
```

Comment:
does this only allow 2 montages, it should scale up to n montages, so avoid iterator named variables like this (make them arrays instead?)

Reply:
this is just to keep track of the two rows of numbers on the left side of the mtg file, although these numbers aren't used for anything right now

it should be able to handle as many montages as specified in the second row of the mtg file, although the whole script will only read a single mtg file

Reply:
yes sorry I ment multiple recordings sessions with single .mtg file

Reply:
so resolve this convo if this is not an issue